### PR TITLE
Fix converter issues

### PR DESCRIPTION
**Pull request by dreab8**

### CODE REVIEW

**Comment by sebersole** [comment]
What is the fix?  Seems like this is trying to deal with exceptions thrown from the converter calls?

**Comment by dreab8** [comment]
@sebersole especially testing that the converter is applied! without the change to `SqmUpdateStatement.java ` the converter is not applied at all, because we resolve the type of the parameter based on the value and not on the attribute, so later on we do not know that we need a converter.

File: hibernate-core/src/main/java/org/hibernate/query/sqm/tree/jpa/AbstractJpaTupleElement.java [hunk 1]
```diff
@@ -47,7 +47,7 @@ protected void setAlias(String alias) {
 		return expressableType;
 	}
 
-	protected final void setExpressableType(SqmExpressable<?> expressableType) {
+	public final void setExpressableType(SqmExpressable<?> expressableType) {
```
Comment (beikov) [inline hibernate-core/src/main/java/org/hibernate/query/sqm/tree/jpa/AbstractJpaTupleElement.java:50]: This change is a bit concerning. Why is it necessary to set the expressable type to null for `JpaCriteriaParameter`? Wouldn't it be better to have some sort of special method on `JpaCriteriaParameter` to take care of this?

**Review by beikov — COMMENTED** [review]
LGTM overall, but the change of visibility of `setExpressableType` is a bit concerning.

**Review by beikov — APPROVED** [review]

**Pull request merged**